contracts/src/v0.8/feeds/DualAggregator.sol

```diff
@@ -816,25 +816,24 @@ contract DualAggregator is OCR2Abstract, OwnerIsCreator, AggregatorV2V3Interface
         _payTransmitter(s_hotVars, report_.juelsPerFeeCoin, uint32(initialGas), msg.sender);
         return;
       }
-      // In case the report doesn't exist, lock the primary feed
-      s_primaryLocked = true;
-    } else {
-      // In case the sender is the primary proxy, unlock the latest report
-      s_primaryLocked = false;
     }
-
     // Report epoch and round
     uint40 epochAndRound = uint40(uint256(reportContext[1]));
 
-    if (epochAndRound < s_hotVars.latestEpochAndRound) {
-      revert StaleReport();
-    }
+    if (epochAndRound != s_hotVars.latestEpochAndRound || !s_primaryLocked) {
+      if (epochAndRound <= s_hotVars.latestEpochAndRound) {
+        revert StaleReport();
+      }
 
-    // Verify signatures attached to report
-    _verifySignatures(reportContext, report, rs, ss, rawVs);
+      // Verify signatures attached to report
+      _verifySignatures(reportContext, report, rs, ss, rawVs);
```

Comment:
Just realized we should always verify signatures. This should be placed under _validateReport

Reply:
Why? To optimize gas costs it's better if we only verify the signatures one time per report.

Reply:
It's an attack vector. You can provide existing reports to the secondary feed that were not signed.

Reply:
Yes, but the existing reports have to have the same timestamp and answer than the one already validated. I thought the biggest benefit of this implementation was to not have to double check the signatures if a report has previously arrived by the other aggregator path.

Reply:
Moved it under `_validateReport()`, so we already have it.